### PR TITLE
Feature/file and bucket name resolution

### PR DESCRIPTION
**Pull request by phutchins**

### CODE REVIEW

File: bin/storj.js [hunk 1]
```diff
@@ -302,25 +333,29 @@ program
...
   .action(utils.changekeyring.bind(program));
 
-  program
-    .command('reset-keyring')
-    .description('delete the current keyring and start a new one')
-    .action(utils.resetkeyring.bind(program));
+program
+  .command('reset-keyring')
+  .alias('rsk')
+  .description('reset the keyring password')
```
Comment (aleitner) [inline bin/storj.js]: change back to old text

File: bin/utils.js [hunk 2]
```diff
@@ -132,6 +132,78 @@ module.exports.getKeyRing = function(keypass, callback) {
   });
 };
 
+module.exports.resolveBucketRef = function(bucketRef, callback) {
```
Comment (aleitner) [inline bin/utils.js]: delete all this

File: bin/actions/files.js [hunk 3]
```diff
@@ -77,10 +83,9 @@ module.exports.mirror = function(bucket, file, env) {
       }
 
       replicas.forEach(function(shard) {
-        log('info', 'Shard %s %s mirroring by %s nodes', [
+        log('info', 'Shard %s mirrored by %s nodes', [
```
Comment (aleitner) [inline bin/actions/files.js]: change back to old log

File: bin/actions/files.js [hunk 4]
```diff
@@ -77,10 +83,9 @@ module.exports.mirror = function(bucket, file, env) {
...
           shard.hash,
-          shard.status,
-          shard.mirrors
+          shard.mirrors.length
```
Comment (bookchin) [inline bin/actions/files.js]: These two lines should be reverted. The new bridge updates do not wait for mirroring acknowledgement, but instead return a 201 and queue up mirroring jobs.

File: package.json [hunk 5]
```diff
@@ -31,6 +31,7 @@
   },
   "dependencies": {
     "storj-lib": "4.0.0-rc.7",
```
Comment (bookchin) [inline package.json]: Someone should bump this to 4.0.0

**Pull request closed without merging**